replica/database.hh

```diff
@@ -412,10 +412,10 @@ private:
 
     compaction_manager& _compaction_manager;
     sstables::compaction_strategy _compaction_strategy;
-    // SSTable set which contains all non-maintenance sstables
-    lw_shared_ptr<sstables::sstable_set> _main_sstables;
     // Holds SSTables created by maintenance operations, which need reshaping before integration into the main set
     lw_shared_ptr<sstables::sstable_set> _maintenance_sstables;
+    // Holds single compaction group owned by table, meaning that all SSTables can be compacted with one another.
+    std::unique_ptr<compaction_group> _compaction_group;
```

Comment:
I suggest placing the member at its final position already in the first patch that introduced it
to reduce the churn.

Right after `_compaction_strategy` seems even more appropriate.

Reply:
I probably needed to put it after maintenance set because compaction_group ctor accessed table::_maintanence_sstables and found it uninitialized. But by the end of the series, _compaction_group will reach that location as those fields are moved into compaction group class.

Reply:
ok, maybe add a FIXME comment to set the stage for that